src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final String welcomeMessage = "Hello! I'm KJ\nHow can I help you today?";
+
+    private static final String completeMessage = "Nice! I've marked this task as done:";
+
+    private static final String byeMessage = "Bye. Hope to see you again soon!";
+
+    private static final String listMessage = "Here are the tasks in your list:";
+
+    private static final int MAXIMUM_TASKS = 100;
+
+    private static boolean shouldContinue;
+
+    private static String command;
+
+    private static String action;
+
+    private static Scanner input;
+
+    private static Task[] instructions;
+
+    private static int count;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        initDuke();
+        displayWelcome();
+        while (shouldContinue) {
+            readInput();
+            setAction();
+            processCommand();
+        }
+    }
+
+    private static void processCommand() {
+        switch(action) {
+        case "bye":
+            sayBye();
+            break;
+        case "list":
+            listTasks();
+            break;
+        case "done":
+            completeTask();
+            break;
+        case "deadline":
+            setDeadline();
+            confirmTask();
+            break;
+        case "event":
+            createEvent();
+            confirmTask();
+            break;
+        case "todo":
+            addTodo();
+            confirmTask();
+            break;
+        }
+    }
+
+    private static void addTodo() {
+        String description = command.substring(5);
+        instructions[count] = new Todo(description);
+        count += 1;
+    }
+
+    private static void createEvent() {
+        int index = command.indexOf("/at");
+        String description = command.substring(6, index - 1);
```

Comment:
Consider avoiding magic numbers when implementing your code as it is not intuitive to the user, who might not be able to understand their purpose. There are multiple uses of magic numbers as indexes in your code which should be avoided.

Reply:
Right, I missed this out when I was removing all the magic strings and numbers. Thanks for pointing out!